### PR TITLE
refresh OS after walls are set to prevent OS error in the objectSensor

### PR DESCRIPTION
**Pull request by GijsGroote**

### CODE REVIEW

File: urdfenvs/urdfCommon/urdf_env.py [hunk 1]
```diff
@@ -270,6 +274,10 @@ def set_walls(self, limits: list = [[-2, -2], [2, 2]]) -> None:
             [0, limits[1][1], 0.0],
             p.getQuaternionFromEuler([0, 0, np.pi / 2]),
         )
+ 
+        # refresh observation space for robots sensors
+        self.refresh_os()
```
Comment (maxspahn) [inline urdfenvs/urdfCommon/urdf_env.py:279]: Why is this needed here? Isn't the function `get_obstacles()` called very often (potentially in every time step)? That would make the simulation very slow I assume.

**Review by maxspahn — COMMENTED** [review]
Nice extracting the functionality into its own function.
Can you elaborate on why the refresh function must be called whenever you call `get_obstacles()`?

File: urdfenvs/urdfCommon/urdf_env.py (hunk 1 above)
Comment (GijsGroote) [reply urdfenvs/urdfCommon/urdf_env.py:279]: This is in the `set_walls()` function. Which adds obstacles and thus should the observation space be refreshed. 

The get_obstacles returns the list of obstacles and is here for completeness. But in the entire project, the function is never used.
